File: DATA-STRUCTURES/Linked_list/Singly_linked_list/Implemented in Java/Deletion.java
import java.io.*;

class Node {
    int item;
    Node next;

    Node() {
        next = null;
    }
}
class LinkedList
{
    Node start;
    public void push() throws IOException {
        System.out.println("Enter Item:");
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int new_item = Integer.parseInt(br.readLine());
        Node temp = new Node();
        temp.item = new_item;
        if (start == null) {
            start = temp;
            return;
        }
        Node p = start;
        while (p.next != null) {
            p = p.next;
        }
        p.next = temp;
        return;
    }
    public void delete_by_item() throws IOException{
        Node p=start;
        Node old=start;
        if (start==null){
            System.out.println("Nothing Io Delete");
            return;
        }   
        System.out.println("ENTER VALUE OF NODE WHICH YOU WISH TO DELETE:");
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int val= Integer.parseInt(br.readLine());
        while (p.item!=val){
            p=p.next;
            if(p==null)
            {
                System.out.println("No such Value Exist");
                return;
            }
        }
        if(p==start)
            start=start.next;
        else if (p.next==null){
            while(old.next.item!=val)
                old=old.next;
            old.next=null;
        }
        else{
            while(old.next.item!=val)
                old=old.next;
            old.next=p.next;
        }
    }
    
    public void delete_by_location() throws IOException {
        Node p = start;
        Node old = start;
        if (start == null) {
            System.out.println("Nothing Io Delete");
            return;
        }
        System.out.println("ENTER LOCATION OF NODE WHICH YOU WISH TO DELETE:");
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int loc = Integer.parseInt(br.readLine());
        if(loc<1)
        {
            System.out.println("INVALID LOCATION");
            return;
        }
        for (int i = 1; i < loc; i++){
            p = p.next;
            if (p == null) {
                System.out.println("No such Location Exist");
                return;
            }
        }
        if (p == start)
            start = start.next;
        else if (p.next == null) {
            for (int i = 1; i < loc-1; i++)
                old = old.next;
            old.next = null;
        } else {
            for (int i = 1; i < loc-1; i++)
                old = old.next;
            old.next = p.next;
        }
    }
    public void deleteLList()
    {
        if(start==null)
        {
            System.out.println("Nothing to delete\n");
            return;
        }
        start=null;
        System.out.println("Linked List Deleted\n");
    }
    public void display() {
        Node p = start;
        System.out.println(" YOUR LINKED LIST LOOKS LIKE :");
        System.out.print("|_HEAD");
        while (p != null) {
            System.out.print(" |_" + p.item + "_|_|->");
            p = p.next;
        }
        System.out.print("[NULL]\n");
    }
}

class Deletion {
    public static void main(String args[]) throws IOException {
        int choice;
        LinkedList obj = new LinkedList();

        while (true) {
            System.out.println("\n LINKED LIST: \n 1. PUSH ELEMENTS FOR DELETION\n 2. DELETE BY VALUE OF NODE\n 3. DELETE BY LOCATION OF NODE\n 4. DELETE LINKED LIST COMPLETELY\n 5. DISPLAY\n 6. EXIT\n\n");
            System.out.println("\nENTER YOUR CHOICE:\n");
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            choice = Integer.parseInt(br.readLine());
            switch (choice) {
            case 1:
                obj.push();
                break;
            case 2:
                obj.delete_by_item();
                break;
            case 3:
                obj.delete_by_location();
                break;
            case 4:
                obj.deleteLList();
                break;

            case 5:
                obj.display();
                break;
            case 6:
                System.exit(0);
                break;
            default:
                System.out.println("\n YOU HAVE ENTERED A WRONG CHOICE\n");
            }
        }
    }
}

File: DATA-STRUCTURES/Linked_list/Singly_linked_list/Implemented in Python/Deletion.py
class Node:
    def __init__(self):
        self.value=None
        Node.next=None
class LinkedList:
    def __init__(self):
        self.start=None

    def append(self):
        item = int(input("Enter Item:\n"))
        newnode = Node()
        newnode.value=item
        if self.start == None:
            self.start= newnode
            return()
        p = self.start
        while p.next != None:
            p = p.next
        p.next = newnode

    def delete_by_item(self):
        p=self.start
        old=self.start
        if self.start==None:
            print("Nothing Io Delete\n")
            return()
        val=int(input("ENTER VALUE OF NODE WHICH YOU WISH TO DELETE:\n"))
        while p.value!=val:
            p=p.next
            if p==None:
                print("No such Value Exist\n")
                return
        if p==self.start:
            self.start=self.start.next
        elif p.next==None:
            while(old.next.value!=val):
                old=old.next
            old.next=None
        else:
            while(old.next.value!=val):
                old=old.next
            old.next=p.next
    
    def delete_by_location(self):
        p=self.start
        old=self.start
        if self.start==None:
            print("Nothing Io Delete\n")
            return()
        loc=int(input("ENTER LOCATION OF NODE WHICH YOU WISH TO DELETE:\n"))
        if loc<1:
            print("INVALID LOCATION")
            return
        for i in range(1,loc):
            p=p.next
            if p==None:
                print("No such Location\n")
                return
        if p==self.start:
            self.start=self.start.next
        elif p.next==None:
            for i in range(1,loc-1):
                old=old.next
            old.next=None
        else:
            for i in range(1,loc-1):
                old=old.next
            old.next=p.next

    def deleteLList(self):
        if self.start==None:
            print("Nothing to Delete")
            return
        p=self.start
        while p!=None:
            nextnode=p.next
            del p
            p=nextnode
        self.start=None
        print("Linked List Deleted\n")

    def display(self):
        p=self.start
        print("YOUR LINKED LIST LOOKS LIKE:\n")
        print("|_HEAD", end="")
        while p != None:
            print("|_", p.value, "_|_|->", end="")
            p=p.next
        print("[NULL]")

exitornot = 1
linkedlist=LinkedList()
while(exitornot):
    print("\n LINKED LIST: \n 1. PUSH ELEMENTS FOR DELETION\n 2. DELETE BY VALUE OF NODE\n 3. DELETE BY LOCATION OF NODE\n 4. DELETE LINKED LIST COMPLETELY\n 5. DISPLAY\n\n")
    choice = int(input("ENTER YOUR CHOICE:\n"))
    switch = {
        1: "append",
        2: "delete_by_item",
        3: "delete_by_location",
        4: "deleteLList",
        5: "display"
    }
    method_name = switch.get(choice)
    if method_name == None:
        print("Invalid Choice")
        continue
    else:
        method=getattr(linkedlist,method_name)
        method()
    exitornot = int(input("Enter 1 to continue and 0 to exit:\n"))

    

